src/CasperClient.cpp

```diff
@@ -108,6 +108,18 @@ GetItemResult Client::GetItem(std::string state_root_hash, std::string key,
                                                    paramsJSON);
 }
 
+/// Returns the item in json at the given address with the given key.
+/// FIXME: It is only a workaround, as GetItem function does not parse
+/// correctly result from json:
+/// https://matterfi.atlassian.net/browse/CD-216
+nlohmann::json Client::GetItem_WA(std::string state_root_hash, std::string key,
+                                  std::vector<std::string> path) {
```

Comment:
why not const ref params?

Reply:
This function has the same interface as `GetItem` function, and I think params are passed by copy in order to have ownership over them. In case this function is run in a multithreaded context and arguments are passed here by immutable reference, then when the thread that has ownership over those params is destroyed and so are the strings/objects that were allocated on it's stack, then we will be left with dangling references.